File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pyramid = new ArrayList<>();

        for (int i=0; i<numRows; i++){
            List<Integer> row = new ArrayList<>();

            row.add(1);
            for (int j=1; j<i; j++){
                int val = pyramid.get(i-1).get(j) + pyramid.get(i-1).get(j-1);
                row.add(val);
            }
            if (i>0) row.add(1);

            pyramid.add(row);
        }

        return pyramid;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return [[]]
        if numRows == 1:
            return [[1]]

        pyramid = [[1]]
        for i in range(1,numRows):
            row = [1] * (i+1)
            for j in range(1,i):
                row[j] = pyramid[i-1][j-1] + pyramid[i-1][j]
            pyramid.append(row)
        
        return pyramid